### ISSUE TITLE
Warning: Encountered known unsupported method torch.instance_norm

### ISSUE THREAD

**Issue by anuar12** [NVIDIA-AI-IOT/torch2trt#177]
Hi, thanks for the great repo!!!
I tried with both installations: with and without plugins from master and got the same warnings.
The conversion ended and speedup was large but the output was very different.

```
Warning: Encountered known unsupported method torch.instance_norm
Warning: Encountered known unsupported method torch.nn.functional.instance_norm
```
Are you planning to support this? Or should I just switch to batch_norm?

**Comment by anuar12** [comment]
I would imagine it would be very similar to `convert_BatchNorm2d` but with repeated scale and bias and changed ScaleMode.ELEMENTWISE.

**Comment by jaybdub** [comment]
Hi anuar12,

Thanks for your feedback!

I agree, it seems like it would be possible to implement with some combination of native TensorRT layers.  Usually this is easier and more distributable than integrating a plugin.   

I'll look into it if I can find time.  If you can easily use batch norm, that may be your safest bet for now.

Best,
John

**Comment by anuar12** [comment]
Great, thanks a promptly advice!

**Comment by kshamaramesh** [comment]
Hi @anuar12 `were able to implement the Instance normalization converter?

**Comment by kshamaramesh** [comment]
@jaybdub  1 ) can we Integrate c ++ plugin of Instance normalization if yes how is it.

Because many are using Instance Normalization. So there should be way of integrating to this

**Comment by jaybdub** [comment]
Hi All,

Thanks for your patience.  I've implemented instance norm in this PR

https://github.com/NVIDIA-AI-IOT/torch2trt/pull/193

Please test it and let me know if it works for your use cases.

I'm also interested in hearing the performance in your application.  I implemented it using a combination of reduction / scaling / elementwise layers.  If there is a performance degredation vs. the plugin, it may be worth integrating a plugin.

Best,
John
